File: src/main/scala/com/abaddon83/burraco/match/games/ports/BurracoGameRepositoryPort.scala
package com.abaddon83.burraco.`match`.games.ports

import com.abaddon83.burraco.`match`.games.domainModels.burracoGames.initialised.BurracoGameInitiatedTurnStart
import com.abaddon83.burraco.`match`.games.domainModels.burracoGames.waitingPlayers.BurracoGameWaitingPlayers
import com.abaddon83.burraco.shares.games.GameIdentity

import scala.concurrent.Future

trait BurracoGameRepositoryPort {

  def save(burracoGame: BurracoGameWaitingPlayers): BurracoGameWaitingPlayers
  def save(burracoGame: BurracoGameInitiatedTurnStart): BurracoGameInitiatedTurnStart
  def findBurracoGameWaitingPlayersBy(gameIdentity: GameIdentity): Future[BurracoGameWaitingPlayers]
  def findBurracoGameInitialisedBy(gameIdentity: GameIdentity): Future[BurracoGameInitiatedTurnStart]
  def findAllBurracoGameWaitingPlayers(): Future[List[BurracoGameWaitingPlayers]]
}

File: src/main/kotlin/com/abaddon83/burracoGame/commandModel/ports/BurracoGameRepositoryPort.kt
package com.abaddon83.burracoGame.commandModel.ports

import com.abaddon83.burracoGame.commandModel.models.BurracoGame
import com.abaddon83.burracoGame.commandModel.models.games.GameIdentity
import com.abaddon83.utils.es.repository.Repository

interface BurracoGameRepositoryPort: Repository<GameIdentity, BurracoGame> {
}